generator.rb

```diff
@@ -0,0 +1,118 @@
+# defining is_vowel? method
+def is_vowel?(letter)
+ vowels = %w[a e i o u]
+ if vowels.include?(letter)
+  return true
+  else
+  return false
+ end
+end
+
+# defining the process to get word_a as a method because it makes the
+# run_generator method itself more clear
+def get_first_word
+  print "Enter your first word:"
+  word_a = gets.chomp.to_s
+
+# setting as false so my until loop will run
+  found_a_vowel = false
+
+# this section will prompt the user to re-enter the word until the
+# length and vowel conditions are met
+  until word_a.length >= 2 && found_a_vowel == true
+    first_word_array = word_a.split('')
+    if first_word_array.any? {|letter| is_vowel?(letter)}
+      found_a_vowel = true
+    else
+      puts "Please enter a word with at least one vowel."
+      word_a = gets.chomp.to_s
+    end
+
+    if word_a.length < 2
+      puts "Please enter a word greater than 1 character long."
+      word_a = gets.chomp.to_s
+    end
+  end
+  # returns the word for run_generator to use later
+  return word_a
+end
+
+
+# defining the process to get word_b as a method because it makes the
+# run_generator method itself more clear
+def get_second_word
+  print "Enter your second word:"
+  word_b = gets.chomp.to_s
+
+# setting as false so my until loop will run
+  found_a_vowel = false
+
+# this section will prompt the user to re-enter the word until the
+# length and vowel conditions are met
+  until word_b.length >= 2 && found_a_vowel == true
+    second_word_array = word_b.split('')
+    if second_word_array.any? {|letter| is_vowel?(letter)}
+      found_a_vowel = true
+    end
```

Comment:
Oh no! Huge bug! What happens if your second word (`word_b`) doesn't have a vowel? To recreate, run your program with the input as first word: `breakfast`, second word: `xx`. It gets stuck in an infinite loop! Why? What's happening? Well, can `found_a_vowel` in this loop ever be `true` in order to exit this loop? I don't think so. I'd like for you to find your bug-- Let me know if you'd like a second pair of eyes to talk this through :)